### ISSUE TITLE
1.13.2 build failed with "Not Found" error for file that exists

### ISSUE THREAD

**Issue by peterse** [tensorflow/tensorflow#31522]
**System information**
- OS Platform and Distribution: Ubuntu 18.04 (Windows 10 subsystem)
- TensorFlow installed from: source
- TensorFlow version: 1.13.2
- Python version: 3.6.8
- Bazel version: 0.21.0
- GCC version: 7.4.0

**Describe the problem**
Just trying to get a CPU hardware-specific install; selected "No" to all except XLA JIT support during configuration. I don't need CUDA/GPU support.

```
ERROR: /mnt/c/Users/peter/Desktop/projects/tensorflow/tensorflow/python/BUILD:5873:1: Executing genrule //tensorflow/python:nccl_ops_pygenrule failed (Aborted): bash failed: error executing command /bin/bash bazel-out/k8-opt/genfiles/tensorflow/python/nccl_ops_pygenrule.genrule_script.sh
2019-08-11 14:35:54.567507: F tensorflow/python/framework/python_op_gen_main.cc:123] Non-OK-status: api_def_map.LoadFileList(env, api_files) status: Not found: tensorflow/core/api_def/base_api/api_def_Imag.pbtxt; No such file or directory
...
Target //tensorflow/tools/pip_package:build_pip_package failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 11735.779s, Critical Path: 3155.14s
INFO: 7082 processes: 7082 local.
FAILED: Build did NOT complete successfully
```

**Provide the exact sequence of commands / steps that you executed before running into the problem**

```
bazel clean --expunge
./configure
bazel build --config=opt //tensorflow/tools/pip_package:build_pip_package
```

**Any other info / logs**

I verified that `tensorflow/core/api_def/base_api/api_def_Imag.pbtxt` does indeed exist.

**Comment by ravikyram** [comment]
Just to verify did you get chance to follow instructions from TensorFlow [website](https://www.tensorflow.org/install/source) .Please, let us know. Thanks!

**Comment by peterse** [comment]
Yes. specifically:
```
git clone https://github.com/tensorflow/tensorflow.git
git checkout v1.13.2
cd tensorflow
```
followed by the build commands from my original post.

**Comment by jvishnuvardhan** [comment]
@peterse What version of nccl and visual studio you are using for the build? Are you installing on Ubuntu or Windows? Thanks!

**Comment by peterse** [comment]
@jvishnuvardhan I am not using nccl (I do not want GPU support) or visual studio; I am installing on an Ubuntu 18.04 Windows Subsystem for Linux (like described at this link: [https://docs.microsoft.com/en-us/windows/wsl/install-win10](https://docs.microsoft.com/en-us/windows/wsl/install-win10)).

This subsystem has been compatibile with every other linux application I've tried to install, so I don't think the subsystem is the problem.

**Comment by mihaimaruseac** [comment]
Can you try with master instead of 1.13?

**Comment by ravikyram** [comment]
@peterse 
Did you get a chance to check with master as suggested by mihaimaruseac. Thanks!

**Comment by peterse** [comment]
This did not resolve the issue. I now get the error:

```
ERROR: /mnt/c/Users/peter/Desktop/projects/tensorflow/tensorflow/python/BUILD:2088:1: Executing genrule //tensorflow/python:batch_ops_pygenrule failed (Aborted): bash failed: error executing command /bin/bash bazel-out/k8-py2-opt/bin/tensorflow/python/batch_ops_pygenrule.genrule_script.sh
2019-08-19 14:49:50.188080: F tensorflow/python/framework/python_op_gen_main.cc:123] Non-OK-status: api_def_map.LoadFileList(env, api_files) status: Not found: tensorflow/core/api_def/base_api/api_def_RandomShuffleQueue.pbtxt; No such file or directory
bazel-out/k8-py2-opt/bin/tensorflow/python/batch_ops_pygenrule.genrule_script.sh: line 2: 19688 Aborted               
```

And again, I verified that this file does indeed exist:
![image](https://user-images.githubusercontent.com/14928221/63291701-11bb1d80-c292-11e9-82b0-79b0010f37d1.png)

Attached: "ERROR" component of failure (there were quite a few Bazel warnings as well...)
[tf_core_dump.txt](https://github.com/tensorflow/tensorflow/files/3517270/tf_core_dump.txt)

**Comment by peterse** [comment]
Also the specs in the original post are now outdated; the relevant changes to versions are:
 - TensorFlow version: 1.14.0 (master)
 - Bazel version: 0.25.2

**Comment by mihaimaruseac** [comment]
I see that the error now points to a different missing file.

Can you confirm that you are running from a fresh clone?

I'm going to try to reproduce

**Comment by mihaimaruseac** [comment]
I was not able to reproduce at all on the latest master, fresh clone:

```shell
/tmp $ git clone https://github.com/tensorflow/tensorflow.git tf
/tmp $ cd tf/
/tmp/tf $ bazel clean --expunge
/tmp/tf $ yes "" | ./configure 
WARNING: Running Bazel server needs to be killed, because the startup options are different.
WARNING: --batch mode is deprecated. Please instead explicitly shut down your Bazel server using the command "bazel shutdown".
You have bazel 0.26.0 installed.
Please specify the location of python. [Default is /usr/bin/python]: 

...
/tmp/tf $ bazel build --config=opt //tensorflow/tools/pip_package:build_pip_package
...
Target //tensorflow/tools/pip_package:build_pip_package up-to-date:
  bazel-bin/tensorflow/tools/pip_package/build_pip_package
INFO: Elapsed time: 2093.317s, Critical Path: 223.10s
INFO: 19513 processes: 19513 local.
INFO: Build completed successfully, 20530 total actions
```

I have bazel 0.26 instead, but doubt this makes a difference. `bazel clean --expunge` at the beginning also should have no effect.

Can you try running the following long command, exactly as written?

```
mkdir tf-new; cd tf-new; git clone https://github.com/tensorflow/tensorflow.git; cd tensorflow; yes "" | ./configure 2>&1 | tee -a github.log; bazel build --config=opt //tensorflow/tools/pip_package:build_pip_package 2>&1 | tee -a github.log
```

Then, if that still fails, can you please attach `github.log` so that we can see if there is another error ahead that causes the behavior you're seeing?

**Comment by peterse** [comment]
I have run the command exactly as you wrote it, and got yet another error claiming that a different file doesn't exist:
```
ERROR: /mnt/c/Users/peter/Desktop/projects/tf-new/tensorflow/tensorflow/python/BUILD:2284:1: Executing genrule //tensorflow/python:ragged_array_ops_pygenrule failed (Aborted): bash failed: error executing command /bin/bash bazel-out/host/bin/tensorflow/python/ragged_array_ops_pygenrule.genrule_script.sh
2019-08-23 14:35:54.149273: F tensorflow/python/framework/python_op_gen_main.cc:123] Non-OK-status: api_def_map.LoadFileList(env, api_files) status: Not found: tensorflow/core/api_def/base_api/api_def_Prelinearize.pbtxt; No such file or directory
```

Though this file does indeed exist:
![image](https://user-images.githubusercontent.com/14928221/63616656-0def0a00-c5b6-11e9-8ec1-b6363e83661f.png)

The log is attached:
[github.log](https://github.com/tensorflow/tensorflow/files/3535816/github.log)

**Comment by peterse** [comment]
I'll also mention that the command you provided defaulted to a python2 installation, while I had previously been running `./configure` and pointing to my python3 executable.

**Comment by mihaimaruseac** [comment]
Start of the log, there are a bunch of errors:

```
ImportError: No module named builtins
```

```
Python Configuration Error: Problem getting numpy include path.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named numpy
```

**Comment by peterse** [comment]
Apologies, those errors were from setting up the environment (I had two failed runs due to `./configure` options using Python 2. Since I am not a Python 2 user, `future` and `numpy` were not installed and since I couldn't locate any explicit Python 2 requirements or a `requirements.txt` I just attempted installation until the build no longer threw import errors. Those errors ended up in the log because of the append mode `tee`).

Please review the revised log where I've removed those irrelevant runs, here:
[github.log](https://github.com/tensorflow/tensorflow/files/3536095/github.log)

**Comment by mihaimaruseac** [comment]
Oh, makes sense.

Seeing warning like the following in the log makes me wonder what compiler are you using?

```
INFO: From Compiling tensorflow/core/protobuf/autotuning.pb.cc [for host]:
In file included from bazel-out/host/bin/tensorflow/core/protobuf/autotuning.pb.cc:4:0:
bazel-out/host/bin/tensorflow/core/protobuf/autotuning.pb.h:234:13: warning: In the GNU C Library, "major" is defined
 by <sys/sysmacros.h>. For historical compatibility, it is
 currently defined by <sys/types.h> as well, but we plan to
 remove this soon. To use "major", include <sys/sysmacros.h>
 directly. If you did not intend to use a system-defined macro
 "major", you should undefine it after including <sys/types.h>.
   ::PROTOBUF_NAMESPACE_ID::int32 major() const;
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from bazel-out/host/bin/tensorflow/core/protobuf/autotuning.pb.cc:4:0:
bazel-out/host/bin/tensorflow/core/protobuf/autotuning.pb.h:240:13: warning: In the GNU C Library, "minor" is defined
 by <sys/sysmacros.h>. For historical compatibility, it is
 currently defined by <sys/types.h> as well, but we plan to
 remove this soon. To use "minor", include <sys/sysmacros.h>
 directly. If you did not intend to use a system-defined macro
 "minor", you should undefine it after including <sys/types.h>.
   ::PROTOBUF_NAMESPACE_ID::int32 minor() const;
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
bazel-out/host/bin/tensorflow/core/protobuf/autotuning.pb.h:374:13: warning: In the GNU C Library, "major" is defined
 by <sys/sysmacros.h>. For historical compatibility, it is
 currently defined by <sys/types.h> as well, but we plan to
 remove this soon. To use "major", include <sys/sysmacros.h>
 directly. If you did not intend to use a system-defined macro
 "major", you should undefine it after including <sys/types.h>.
   ::PROTOBUF_NAMESPACE_ID::int32 major() const;
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
bazel-out/host/bin/tensorflow/core/protobuf/autotuning.pb.h:380:13: warning: In the GNU C Library, "minor" is defined
 by <sys/sysmacros.h>. For historical compatibility, it is
 currently defined by <sys/types.h> as well, but we plan to
 remove this soon. To use "minor", include <sys/sysmacros.h>
 directly. If you did not intend to use a system-defined macro
 "minor", you should undefine it after including <sys/types.h>.
   ::PROTOBUF_NAMESPACE_ID::int32 minor() const;
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Also, could it be possible you're running out of space? Timeout? Corrupted disk? Can you try compiling on a Linux only box or directly on Windows instead of WSL? Can you reinstall the WSL?

When the compilation ends in error, do you have the bazel-* symlinks?

**Comment by peterse** [comment]
Found out the problem: it was a multithreading bug from the _host_ Windows OS that causes these kinds of 'file not found' errors in WSL if using Windows 10 OS versions before 17655.

For future reference, the problem can be resolved by updating Windows 10 to some version after 17655; thread detailing this issue [here](https://stackoverflow.com/questions/50206890/intermittent-random-file-not-found-errors-under-windows-subsystem-for-linux). After this I was able to build and install successfully.
